lib/tag-input/index.tsx

```diff
@@ -175,11 +195,14 @@ export class TagInput extends Component<Props> {
     });
   };
 
-  onInput = (value: string) => {
+  onInput = (value: string, { moveCaretToEndOfValue = false } = {}) => {
     if (this.state.isComposing) {
       return;
     }
 
+    this.caretPosition = moveCaretToEndOfValue
+      ? value.length
+      : getCaretPosition(this.inputField);
```

Comment:
I don't find the addition of `this.caretPosition` or `moveCaretToEndOfValue` to be the most elegant solution. However, given the large number of edge cases for different types of input, keystrokes, clicks, etc, it was the only one I could find that covered all the cases. I definitely welcome feedback or thoughts on this approach.